### DISCUSSION TITLE
Future support

### DISCUSSION THREAD

**Discussion by Svardsten53** [mcguirepr89/BirdNET-Pi#1003] [General]
Has **ehpersonal38** taken a break or left the project? It is now a couple of months since we saw any updates or the excellent support that users received in case of problems.

Currently I have no need for support, my systems work very well, but there are some questions asked by other users. In my opinion the program doesn't lack any functions either, but I'm a little confused about how future updates to the analysis model will be activated if no administrator is available.

**Comment by srd424** [comment]
Would be good to know - I've got various patches and PRs floating around, some basically just bug fixes, some add small new options / features, others are a bit more architectural and the sort of thing that it would be good to discuss with upstream to see if they're acceptable or not.

On the other hand, I do understand that "real life" has a habit of intruding upon and displacing hobbies, side projects, etc :(

I keep meaning to see if learning stgit can help me manage stacks of patches, but I'm not firing on all cylinders right now sadly.

**Comment by ehpersonal38** [comment]
Yep, I've taken an indefinite break from this project, but I will stlll try to support it with critical fixes and model updates. I'll only be accepting PRs that fix bugs from now on, in order to keep the project fairly stable.

**Reply by srd424** [reply]
Ah, that's good to know - feel free to close off my more adventurous stuff (or I'll do it later), I will get my act together and get my own repo tidied up a bit :)

#970 and #975 are sort of borderline bug fixes:

* #970 avoids data loss if the root filesystem is fuller than the filesystem mounted over `~pi/BirdSongs`, and hopefully is 'obviously correct.' Even outside of a container environment, I guess someone might use an external hard drive or NFS mount even on an RPi.
*   #975 is tidy-up but might make it easier / less confusing to fix things like the default model apparently being out-of-date?

**Reply by Svardsten53** [reply]
It feels very good that you are keeping an eye on the system in the future as well. I am really very grateful for all the work you have put in so far.

**Comment by lloydbayley** [comment]
I've had a few messages about support for birdnetpi so i'll reply here so it might help. I've given this project a miss over the past while because it's turned into a big pile of change requests that are extra to what was originally intended. This makes it more and more impossible to keep up with. Once the stuff started becoming beyond bugs, it was turning into an 'i want this' sort of thing, which is fine but once it extends beyond the original intent of the project and becomes a re-write, then it becomes almost impossible to support. The original owner of the repository has all but abandoned it due to health reasons. Any control, be it triage or contributor is not forthcoming and has been met with resistance, so I have given up. I've tried, god knows I've tried but I suggest that anyone wanting to continue this project forks their own and continues under their own steam. This might clarify things and address the messages I have had regarding why I am not helping with birdnetpi anymore. I'm not following the repo or subscribe to messages from it. All the best.

**Reply by Svardsten53** [reply]
I agree. During the two years of its existence, the program has made great progress with many good features. In my opinion, nothing is missing in the functionality. Since support is now very limited, there is no reason to compromise stability by introducing a lot of new features. If an update to the analysis model comes in the future, it should of course be included if possible, but it is only maintenance of an existing function.

**Reply by ehpersonal38** [reply]
Agree with all the above. Honestly this project has taught me a lot about software management. I have learned that there comes a time where more changes just start to detract from the software and muddle everything rather than improve. Knowing when to _not_ develop is as important as the development itself. If I were to continue working on BirdNET-Pi I'd probably strip away everything non-essential, make the core elements as stable as possible, then maybe re-package all the QOL stuff we have (Apprise, Flickr, all the neat Settings, etc) into their own modules/plugins that would never have the possibility of breaking the main code. It'd be more work but it would avoid the cluster-f that this project has turned into. I don't mean that out of malice or anything, I love the idea of this project and am very grateful for everyone in the community over the past few years. I hope it can turn into something better, or something better can use it as inspiration to take its place. 🙂

**Reply by Vertikar** [reply]
> If I were to continue working on BirdNET-Pi I'd probably strip away everything non-essential, make the core elements as stable as possible

I feel like this is sort of the path that @mmcc-xx has gone down with BirdCAGE - https://github.com/mmcc-xx/BirdCAGE
Unfortunately they have also been unable to continue development at this stage.

I'm currently spinning up another RPi for a portable remote deployment and I'm playing around with BirdCAGE to see if I'm happy with it, and may consider moving my main home setup to that in the future. It's also dockerised, so easy to deploy on a server, and potentially have a microphone deployed remotely on an ESP32 or a RPI Zero

**Reply by tphakala** [reply]
I am working on a Golang conversion of BirdNet Analyzer, if I ever get it working plan is to work in into BirdNet-Pi type solution with power efficiency in mind to support battery powered listening stations.

**Reply by srd424** [reply]
> I'm currently spinning up another RPi for a portable remote deployment and I'm playing around with BirdCAGE to see if I'm happy with it, and may consider moving my main home setup to that in the future. It's also dockerised, so easy to deploy on a server, and potentially have a microphone deployed remotely on an ESP32 or a RPI Zero

I've got basically that set-up working - dockerized BirdNet-Pi on an x86-64 server, with gstreamer on a Pi Zero chucking audio at it. 
Unfortunately I've also been distracted by real world problems! BirdNet-Pi's user interface is nice, it may look a little dated to youngsters but it provides a fair amount of different views onto the data. I did think patching it to make it talk to the upstream version of BirdNet-Analyzer would perhaps prolong its life.

**Reply by tphakala** [reply]
Got the Golang version of analyzer working, I will push it to gitlab later this week.

thakala@ubuntu:~/go/src/github.com/tphakala/birdnet-go$ go run main.go 
Starting BirdNet
- Reading audio data
- Done
- Read 5 chunks
- Analyzing audio data
- Done
Time 1.173592 seconds

Time Interval: 12.000000;15.000000	Strix aluco_Tawny Owl: 0.871380
Time Interval: 0.000000;3.000000	Strix aluco_Tawny Owl: 0.975716
Time Interval: 3.000000;6.000000	Strix aluco_Tawny Owl: 0.903863
Time Interval: 6.000000;9.000000	Strix aluco_Tawny Owl: 0.983856
Time Interval: 9.000000;12.000000	Strix aluco_Tawny Owl: 0.270588
thakala@ubuntu:~/go/src/github.com/tphakala/birdnet-go$

**Reply by srd424** [reply]
Is it noticeably more efficient than the python version? I sort of assumed the vast majority of the CPU time would be spent within tflite itself. Still, a static compiled binary would be nice for deployment and containerization purposes :)

**Reply by tphakala** [reply]
I can’t say about performance yet, but my main motivation is to get rid of depencency mess of python

**Reply by tphakala** [reply]
Go-BirdNet is now up at https://github.com/tphakala/go-birdnet, it is very much of work in progress and does not yet live up to promise of ease of deployment as you need to have tflite source available to build

**Reply by tphakala** [reply]
I now have a working realtime detection built in, this works fully within memory only at the moment. Sound is captured into 3 second buffer which is polled by a buffer monitor which then feeds data to BirdNet engine for detection. Output is still just printing to stdout but I will add SQLite backend and work out saving interesting sound clips to disk.

**Reply by tphakala** [reply]
I just released a new build of BirdNET-Go, it contains lots of fixes for previous versions.

https://github.com/tphakala/birdnet-go/releases/tag/v0.4.4

Backend side is starting to be on good level, I have been using this for few months already, I am also using it with my winter time birdfeeder stream at https://www.youtube.com/watch?v=FXW1efd9Dro

Web UI is still at very early stages and not ready for release.

**Reply by tphakala** [reply]
I have now published code with Web UI to birdnet-go github, web ui is still very early stages but it can be used to view daily stats can edit birdnet-go configuration. If anyone is willing to help I am open for contributions! Web UI part is based on Go HTML templates, HTMX and Tailwind CSS.

![Web capture_5-2-2024_203234_192 168 4 19](https://github.com/mcguirepr89/BirdNET-Pi/assets/7030001/b30e2cda-8af9-4d7a-9d09-59d1b81bb90d)
